target/linux/ath79/dts/qca9558_mikrotik_routerboard-922uags-5hpacd.dts

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "mikrotik,routerboard-922uags-5hpacd", "qca,qca9558";
+	model = "MikroTik RouterBOARD 922UAGS-5HPacD";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-upgrade = &led_user;
+		serial0 = &uart;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+
+		led_user: user {
+			label = "mikrotik:green:user";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	ath10k-leds {
+		compatible = "gpio-leds";
+
+		wlan5g {
+			label = "mikrotik:green:wlan5g";
+			gpios = <&ath10k 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_usb_power {
+			gpio-export,name = "mikrotik:power:usb";
+			gpio-export,output = <0>;
+			gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_nand_power {
+			gpio-export,name = "mikrotik:power:nand";
+			gpio-export,output = <0>;
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
```

Comment:
Please double-check these two, as I'm too stupid to translate the high/low correctly from ar71xx.

Reply:
The NAND one is weird:
- ACTIVE_LOW, OUTPUT 0: NAND OK
- ACTIVE_LOW, OUTPUT 1: NO NAND
- ACTIVE_HIGH, OUTPUT 0: NAND OK
- ACTIVE_HIGH, OUTPUT 1: NO NAND

The USB one is fine, I can turn a USB dongle ON and OFF.

Reply:
No idea.

Reply:
I've just learned that the only relevant option for setting the value is "output". GPIO_ACTIVE_* should then by chosen according to when the setting is enabled, but it has no effect on which value is set.

Reply:
Don't know whether the different values for GPIO_ACTIVE are correct here. But I'm still wocking on fully understanding this stuff...